src/python/module/smartredis/util.py

```diff
@@ -111,3 +114,136 @@ def typecheck(arg, name, _type):
+
+                if isinstance(eval(arg),list):
+                    items = []
+                    for item in eval(arg): 
+                        items.append(item)
+
+                    arg_field=",".join(items)
+                    ds.add_meta_string(f"_xarray_{d}_{arg}",arg_field)
+            
+                else:
+                    if eval(arg) != None: 
+                        ds.add_meta_string(f"_xarray_{d}_{arg}",eval(arg))
+                        print(f"_xarray_{d}_{arg}")
+    
+                    else: 
+                        ds.add_meta_string(f"_xarray_{d}_{arg}","null")
+    
+        
+     
+    def transform_to_xarray(ds): 
+        """ Uses dataformat specifc field names added to the 
+        SmartRedis dataset in the add_metadata method to construct 
+        xarray in transform method 
+
+        :param ds: a SmartRedis dataset that has had  metadata added 
+        to dataformat specificfield names 
+        :type ds: SmartRedis Dataset
+        
+        :return: a dictionary of keys as the data field anme and the
+        value as the built Xarray DataArray constructed using 
+        fieldnames and appropriately formatted metadata. 
+        """
+        
+        coord_dict= {}
+        coord_final = {}
+        attr_dict = {}
+        
+        
+        iter_ = ds.get_meta_strings("_xarray_data_names")
+  
+        for item in ds.get_meta_strings("_xarray_data_names"):
+       
+            # make the coordinate dict 
+            for d in ds.get_meta_strings("_xarray_data_names"):
+                for k in ds.get_meta_strings(f"_xarray_{item}_coord_names")[0].split(","):
+                    if d == k:
+                        
+                        if d in iter_:
+                            iter_.remove(d)
+              
+                        # coordinate dimensions
+                        test_dims_final = []
+                        for qitem in ds.get_meta_strings(f"_xarray_{d}_dim_names")[0].split(","):
+                            test_xarray_dims = ds.get_meta_strings(qitem)
+   
+                            test_dims_final.append(test_xarray_dims[0])
+                        
+                        # coordinate attributes 
+                        for atitem in ds.get_meta_strings(f"_xarray_{d}_attr_names")[0].split(","):
+                            
+                            attrt = ds.get_meta_strings(atitem)
+                            attr_dict[atitem] = attrt[0] 
+                                            
+                        coord_dict[d]= (test_dims_final,ds.get_tensor(d),attr_dict)
+        
+                        # clear the coordinate attributes dict 
+                        attr_dict = {}
+                
+                    # dict: keys= data names, value = their coordinate dictionaries 
+                        coord_final[item] = coord_dict
+                        
+        # rest of the data items (not coordinates)                
```

Comment:
Add a verb here? "Loop through the rest of the data variables and construct xr.DataArrays